config.go

```diff
@@ -0,0 +1,324 @@
+}
+
+type TestHosts struct {
+	URLs         []string
+	PingWaitTime time.Duration `mapstructure:"ping-wait-time"`
+	PingRetries  int           `mapstructure:"ping-retries"`
+}
+
+var DefaultTestHosts = TestHosts{
+	URLs:         []string{"1.1.1.1", "8.8.8.8"},
+	PingWaitTime: time.Second * 30,
+	PingRetries:  5,
+}
+
+type Battery struct {
+	EnableVoltageReadings bool   `mapstructure:"enable-voltage-readings"`
+	NoBattery             uint16 `mapstructure:"no-battery-reading"`
+	LowBattery            uint16 `mapstructure:"low-battery-reading"`
+	FullBattery           uint16 `mapstructure:"full-battery-reading"`
+}
+
+type Modemd struct {
+	TestInterval      time.Duration `mapstructure:"test-interval"`
+	InitialOnDuration time.Duration `mapstructure:"initial-on-duration"`
+	FindModemTimeout  time.Duration `mapstructure:"find-modem-timeout"`
+	ConnectionTimeout time.Duration `mapstructure:"connection-timeout"`
+	RequestOnDuration time.Duration `mapstructure:"request-on-duration"`
+	Modems            []modem       `mapstructure:"modems"`
+}
+
+type modem struct {
+	Name            string `mapstructure:"name"`
+	NetDev          string `mapstructure:"net-dev"`
+	VendorProductID string `mapstructure:"vendor-product-id"`
+}
+
+var DefaultModemd = Modemd{
+	TestInterval:      time.Minute * 5,
+	InitialOnDuration: time.Hour * 24,
+	FindModemTimeout:  time.Minute * 2,
+	ConnectionTimeout: time.Minute,
+	RequestOnDuration: time.Hour * 24,
+	Modems: []modem{
+		modem{Name: "Huawei 4G modem", NetDev: "eth1", VendorProductID: "12d1:14db"},
+		modem{Name: "Spark 3G modem", NetDev: "usb0", VendorProductID: "19d2:1405"},
+	},
+}
+
+type Lepton struct {
+	SPISpeed    int64  `mapstructure:"spi-speed"`
+	FrameOutput string `mapstructure:"frame-output"`
+}
+
+var DefaultLepton = Lepton{
+	SPISpeed:    2000000,
+	FrameOutput: "/var/run/lepton-frames",
+}
+
+type ThermalRecorder struct {
+	OutputDir      string `mapstructure:"output-dir"`
+	MinDiskSpaceMB uint64 `mapstructure:"min-disk-space-mb"`
+	MinSecs        int    `mapstructure:"min-secs"`
+	MaxSecs        int    `mapstructure:"max-secs"`
+	PreviewSecs    int    `mapstructure:"preview-secs"`
+}
+
+var DefaultThermalRecorder = ThermalRecorder{
+	MaxSecs:        600,
+	MinSecs:        10,
+	PreviewSecs:    3,
+	MinDiskSpaceMB: 200,
+	OutputDir:      "/var/spool/cptv",
+}
+
+type ThermalMotion struct {
+	DynamicThreshold bool   `mapstructure:"min-secs"`
+	TempThresh       uint16 `mapstructure:"temp-thresh"`
+	DeltaThresh      uint16 `mapstructure:"delta-thresh"`
+	CountThresh      int    `mapstructure:"count-thresh"`
+	FrameCompareGap  int    `mapstructure:"frame-compare-gap"`
+	UseOneDiffOnly   bool   `mapstructure:"use-one-diff-only"`
+	TriggerFrames    int    `mapstructure:"trigger-frames"`
+	WarmerOnly       bool   `mapstructure:"warmer-only"`
+	EdgePixels       int    `mapstructure:"edge-pixels"`
+	Verbose          bool   `mapstructure:"verbose"`
+}
+
+var DefaultThermalMotion = ThermalMotion{
```

Comment:
Please make all the defaults be created by a function rather than having a mutable global. I've seen plenty of obscure bugs caused by global default structs being accidentally modified.